### PR TITLE
Update eslint to v5

### PR DESCRIPTION
**Pull request by jackharrhy**
# What
- `eslint` v4 -> v5, hence a major update will be required later
- Other deps. upgraded with `ncu`
- `experimentalObjectRestSpread` deprecated, now uses `ecmaVersion: 2018` instead
- `no-await-in-loop` ignored
- `ecmascript-9` rules and configuration added
- new rules added, other deprecated rules removed
- `test.sh` edited to allow for commands like `npm run lint -- --fix`
# Test
- `npm run test` passes
- link it to an existing project, other than any new rules everything should still work

### CODE REVIEW

File: rules/ecmascript-6.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ module.exports = {
 		'prefer-rest-params': 'error',
 		'prefer-spread': 'error',
 		'prefer-template': 'error',
+		'prefer-object-spread': 'error',
```
Comment (MitMaro) [inline rules/ecmascript-6.js]: This is under the wrong category. But it's also not supported before es2018, so we should add it to a `ecmascript-9` rules file and only include it in the ecmascript-9 config.

File: rules/plugin-node.js [hunk 2]
```diff
@@ -9,8 +9,15 @@ module.exports = {
...
 		'node/no-unpublished-require': 'off',
-		'node/no-unsupported-features': 'error',
+		'node/no-unsupported-features/es-syntax': 'error',
+		'node/no-unsupported-features/es-builtins': 'error',
+		'node/no-unsupported-features/node-builtins': 'error',
 		'node/process-exit-as-throw': 'error',
+		'node/prefer-global/buffer': 'error',
+		'node/prefer-global/console': 'error',
+		'node/prefer-global/url': ['error', 'never'],
+		'node/prefer-global/url-search-params': ['error', 'never'],
```
Comment (MitMaro) [inline rules/plugin-node.js]: Let's just turn off the rules for url and url-search-pattern for now. That way Node 10 code can start using the globals.

Comment (jackharrhy) [reply rules/plugin-node.js]: 67a9feb3cc4c29b087199ea05444b6f7f8e3df48

File: rules/ecmascript-6.js (hunk 1 above)
Comment (jackharrhy) [reply rules/ecmascript-6.js]: 67a9feb3cc4c29b087199ea05444b6f7f8e3df48

File: config/react.js [hunk 3]
```diff
@@ -9,9 +9,9 @@ module.exports = {
 	},
 	parserOptions: {
 		sourceType: 'module',
+		ecmaVersion: 2018,
```
[the remark this answers is not in the document]
Comment (jackharrhy) [reply config/react.js]: 67a9feb3cc4c29b087199ea05444b6f7f8e3df48

**Comment by MitMaro** [comment]
Let's get this merged, we can make a few other changes after if needed.

**Comment by MitMaro** [comment]
@jackharrhy Would you mind rebasing the changes, and run `ncu` again to update to the newest dependencies?

**Comment by jackharrhy** [comment]
Updated at 043f057d1c5ce4c56cd12f67d518a45839ccdd28 @MitMaro

**Pull request merged**